File: code/mathematical_algorithms/src/gcd_and_lcm/gcd_and_lcm.cpp
#include <stdio.h>
#include <algorithm>
// Part of Cosmos by OpenGenus Foundation
int gcd(int x, int y)
{
    while (y > 0)
    {
        x %= y;
        std::swap(x, y);
    }
    return x;
}

//should use this recursive approach.
// int gcd(int c,int d)
// {
//     if(d==0)
//         return a;
//     return gcd(d,c%d);
// }

int lcm(int x, int y)
{
    return x / gcd(x, y) * y;
}

int main()
{
    int a, b;
    scanf("%d %d", &a, &b);
    printf("GCD = %d\n", gcd(a, b));
    printf("LCM = %d", lcm(a, b));
}

File: code/mathematical_algorithms/src/gcd_and_lcm/gcd_and_lcm.js
function gcd(a, b) {
  return b === 0 ? a : gcd(b, a % b);
}

function lcm(a, b) {
  return b === 0 ? 0 : (a * b) / gcd(a, b);
}

// GCD
console.log(gcd(15, 2)); // 1
console.log(gcd(144, 24)); // 24

// LCM
console.log(lcm(12, 3)); // 12
console.log(lcm(27, 13)); // 351
